src/libraries/Native/Unix/System.Globalization.Native/pal_icushim_static.c

```diff
@@ -0,0 +1,80 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+//
+
+#include <stdlib.h>
+#include <stdio.h>
+#include "pal_icushim_internal.h"
+#include "pal_icushim.h"
+#include <unicode/putil.h>
+#include <unicode/uversion.h>
+#include <unicode/localpointer.h>
+#include <unicode/utrace.h>
+
+static void log_icu_error (const char * name, UErrorCode status) {
+    const char * statusText = u_errorName(status);
+    fprintf(stderr, "ICU call %s failed with error #%d '%s'.\n", name, status, statusText);
+}
+
+static void U_CALLCONV icu_trace_data (const void *context, int32_t fnNumber, int32_t level, const char *fmt, va_list args) {
+    char buf [1000];
+    utrace_vformat (buf, sizeof (buf), 0, fmt, args);
+    printf ("[ICUDT] %s: %s\n", utrace_functionName (fnNumber), buf);
+}
+
+#ifdef __EMSCRIPTEN__
+#include <emscripten.h>
+
+EMSCRIPTEN_KEEPALIVE int32_t mono_wasm_load_icu_data (void * pData);
+
+EMSCRIPTEN_KEEPALIVE int32_t mono_wasm_load_icu_data (void * pData) {
+    UErrorCode status = 0;
+    udata_setCommonData (pData, &status);
+
+    if (U_FAILURE(status)) {
+        log_icu_error("udata_setCommonData", status);
+        return 0;
+    } else {
+        //// Uncomment to enable ICU tracing,
+        //// see https://github.com/unicode-org/icu/blob/master/docs/userguide/icu_data/tracing.md
+        // utrace_setFunctions(0, 0, 0, icu_trace_data);
+        // utrace_setLevel(UTRACE_VERBOSE);
+        return 1;
+    }
+}
+#endif
+
+int32_t GlobalizationNative_LoadICU(void)
+{
+    const char* icudir = getenv("DOTNET_ICU_DIR");
+    if (icudir)
+        u_setDataDirectory(icudir);
+    else
+        ; // default ICU search path behavior will be used, see http://userguide.icu-project.org/icudata
+
+    UErrorCode status = 0;
+    // Invoking u_init will probe to see if ICU common data is already available and if it is missing,
```

Comment:
FWIW, just wanted to add a note/comment: 🙂

If the ICU data file might not have any code-pages (ICU calls them converters or mappings), because they got filtered out for space concerns, then the call to `u_init` might fail.
(Internally, the `u_init` function just tries to read the number of code-pages in the data file)

Instead, you could possibly consider calling the function `ulocdata_getCLDRVersion`, which will attempt to read the CLDR version from the ICU data file (which is really small and should likely always be there).

The file `pal_icushim.c` already does this here, so it might also be one less API to pull in as well:
https://github.com/dotnet/runtime/blob/c0fb094d7237b43a814249f5ee2b3223f1d0884f/src/libraries/Native/Unix/System.Globalization.Native/pal_icushim.c#L369

There a bit more context/background in one of the comment threads on the PR here:
https://github.com/dotnet/runtime/pull/35383#discussion_r418770620

Reply:
Good observation. I see from your comment that people strip the code-page data from the packages in the real world, so we should probably use getCLDRVersion.

Reply:
ulocdata_getCLDRVersion doesn't work, but u_init does. It's not clear to me why.
```ulocdata_getCLDRVersion failed with error #2 'U_MISSING_RESOURCE_ERROR'```
Do you know what specific bit of the data file needs to be present for it to work?

Reply:
Hmm, that's a bit odd... I wonder if maybe it could be getting filtered out perhaps?

The piece of data that that it uses/loads comes from this:
`cldrVersion{"37"}`
Link to the raw data:
https://github.com/microsoft/icu/blob/master/icu/icu4c/source/data/misc/supplementalData.txt#L2107

Do you have the built/compiled data file available somewhere? (I could try to open/dump it out and see if it got removed perhaps...).

Reply:
There is an icudt.dat in the microsoft.netcore.runtime.icu.transport/5.0.0-preview.8.20364.1 nuget package. Here's a copy of it https://1drv.ms/u/s!AiLk0W5QUEJHoyXyQb9Pp8Wo1iuM?e=XhGycD

Reply:
Thanks for the link/copy!
Dumping out the contents I don't see the "misc" folder at all. (This would explain why the API fails, as there's no data for it).

```
brkitr/
coll/
curr/
lang/
region/
```
Do you have a link for the filter that was used?

Reply:
FWIW, the `u_init` API still works as it is looking for `cnvalias.icu` which still exists in the data file.

Since you don't have any converters/code-pages at all, you could possibly remove it (assuming you can get the `ulocdata_getCLDRVersion` API working), which would save ~63K. :)

I think you could do this with:
```json
"featureFilters": {
  "cnvalias": { "filterType": "exclude" },
```

Reply:
@jefgen thanks, resolved via https://github.com/dotnet/icu/pull/4